### PR TITLE
Move logging initialization into Fastly adapter (PR 10)

### LINKED ISSUE #491: Abstract logging initialization
## Description

Move `log-fastly` initialization to the Fastly adapter. Core crate uses only `log` macros with no Fastly-specific backend.

## Done when

- `log-fastly` in adapter only; core uses `log` macros
- Per-PR gates pass

## Blocked by

PR 2

## References

- `docs/internal/EDGEZERO_MIGRATION.md` — Phase 1, PR 10
- Part of #480

### PR DESCRIPTION
**Pull request by prk-Jr**
## Summary

- Move Fastly logging initialization into an adapter-local module so backend setup stays owned by the Fastly adapter instead of leaking through `main.rs`.
- Remove `log-fastly` from `trusted-server-core` so core remains backend-agnostic and future Edgezero adapters can rely on the shared `log` facade only.
- Capture the PR 10 design and implementation notes used to keep this stacked migration step scoped against the PR 9 base branch.

## Changes

| File | Change |
| ---- | ------ |
| `crates/trusted-server-adapter-fastly/src/logging.rs` | Add the adapter-local Fastly logger initializer, target-label helper, and unit test. |
| `crates/trusted-server-adapter-fastly/src/main.rs` | Call `logging::init_logger()` and remove the inline Fastly logger implementation. |
| `crates/trusted-server-core/Cargo.toml` | Remove the `log-fastly` dependency from core. |
| `Cargo.lock` | Update the lockfile after removing core's `log-fastly` dependency edge. |
| `docs/superpowers/specs/2026-04-02-pr10-logging-initialization-design.md` | Add the PR 10 design note and scope. |
| `docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md` | Add the implementation plan for the logging boundary change. |

## Closes

Closes #491

## Test plan

- [x] `cargo test --workspace`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo fmt --all -- --check`
- [x] JS tests: `cd crates/js/lib && npx vitest run`
- [x] JS format: `cd crates/js/lib && npm run format`
- [x] Docs format: `cd docs && npm run format`
- [ ] WASM build: `cargo build --package trusted-server-adapter-fastly --release --target wasm32-wasip1`
- [ ] Manual testing via `fastly compute serve`
- [ ] Other:

## Hardening note

Not applicable. This PR does not touch config-derived regex or pattern compilation.

## Checklist

- [x] Changes follow [CLAUDE.md](/CLAUDE.md) conventions
- [x] No `unwrap()` in production code — use `expect("should ...")`
- [x] Uses `log` macros (not `println!`)
- [x] New code has tests
- [x] No secrets or credentials committed

### CODE REVIEW

**Review by aram356 — APPROVED** [review]
## Summary

Clean, well-scoped extraction that correctly moves Fastly logging initialization into the adapter, making `trusted-server-core` backend-agnostic. No behavioral changes.

## Non-blocking

### 🔧 wrench

- **Unnecessary `#[allow(dead_code)]`**: `init_logger()` is called from `main.rs` — the suppression is incorrect and would mask a real warning if the call is removed (`logging.rs:8`)

### 🤔 thinking

- **`_message` vs `message`**: The original used the `message` parameter directly (idiomatic fern). The rename to `_message` + `record.args()` is semantically identical but gratuitous (`logging.rs:18`)

### 🌱 seedling

- **`rsplit_once` alternative**: More explicit than `split("::").last()` for extracting the final segment (`logging.rs:4`)
- **Missing test edge cases**: No coverage for inputs without `::`, empty strings, or trailing `::` (`logging.rs:38`)

### 👍 praise

- **Clean separation of concerns**: `log-fastly` correctly removed from core, adapter owns its logging, `pub(crate)` visibility is correct
- **Thorough design and plan docs**: Clear scope, out-of-scope boundaries, and reproducible implementation plan

**Review by ChristianPavilonis — APPROVED** [review]

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+use chrono::{Local, SecondsFormat};
+use log_fastly::Logger;
+
+pub(crate) fn target_label(target: &str) -> &str {
+    target.rsplit_once("::").map_or(target, |(_, last)| last)
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: 🤔 **thinking** — `rsplit_once("::")` on `"trailing::"` returns `""`, which would produce log output like `INFO [] message`. Real `log` targets won't hit this, but consider falling back to the full target when the result is empty:

```rust
pub(crate) fn target_label(target: &str) -> &str {
    target
        .rsplit_once("::")
        .map_or(target, |(_, last)| if last.is_empty() { target } else { last })
}
```

File: docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+- [ ] **Step 3: Implement the minimal helper and adapter logger initializer**
+
+Replace the skeleton with the real adapter-local module:
+
+```rust
+use chrono::{Local, SecondsFormat};
+use log_fastly::Logger;
+
+pub(crate) fn target_label(target: &str) -> &str {
+    target.split("::").last().unwrap_or(target)
```
Comment (aram356) [inline docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md]: ⛏ **nitpick** — Plan shows `split("::").last().unwrap_or(target)` but the implementation uses `rsplit_once`. The implementation is better (avoids iterating all segments), but the plan doc is now inaccurate.

**Review by aram356 — CHANGES_REQUESTED** [review]
## Summary

Clean extraction of Fastly logging initialization from `trusted-server-core` into `trusted-server-adapter-fastly`. The change is small, well-scoped, and achieves the stated goal of making core backend-agnostic for logging. CI passes.

## Non-blocking

### 🤔 thinking

- **`target_label` empty-result fallback**: `rsplit_once` on `"trailing::"` returns `""`, producing log output like `INFO [] message`. Consider falling back to the full target when the extracted segment is empty. (crates/trusted-server-adapter-fastly/src/logging.rs:5)

### 📌 out of scope

- **`trusted-server-core` still depends on `fastly` directly**: Removing `log-fastly` is a step toward backend-agnostic core, but `fastly = { workspace = true }` remains in core's Cargo.toml. Worth tracking as a follow-up. (crates/trusted-server-core/Cargo.toml:25)

### 🌱 seedling

- **Missing doc comments on `target_label` and `init_logger`**: Both are `pub(crate)` and lack `///` comments. `init_logger` is a key startup function and `target_label` has non-obvious semantics — a brief doc comment would help. (crates/trusted-server-adapter-fastly/src/logging.rs:4,8)

### ⛏ nitpick

- **Plan/implementation mismatch**: Plan shows `split("::").last().unwrap_or(target)` but implementation uses `rsplit_once`. Implementation is better, but the plan doc is now inaccurate. (docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md:81)

## CI Status

- integration tests: PASS
- browser integration tests: PASS
- prepare integration artifacts: PASS

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
+use chrono::{Local, SecondsFormat};
+use log_fastly::Logger;
+
+/// Extracts the final `::` segment from a Rust module path for use as a log label.
+///
+/// Falls back to the full target string when the input contains no separator or
+/// when the separator appears at the trailing position (e.g. `"foo::"`), which
+/// would otherwise produce an empty label in log output.
+pub(crate) fn target_label(target: &str) -> &str {
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: ⛏ **nitpick** — `target_label` is only used inside this module, so `pub(crate)` is broader than needed. A private `fn target_label` would still be reachable from `#[cfg(test)] mod tests`. Trivial, feel free to ignore.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    target.rsplit_once("::").map_or(
+        target,
+        |(_, last)| if last.is_empty() { target } else { last },
+    )
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: ⛏ **nitpick** — On the theoretical `"trailing::"` input, the fallback returns the full input including the `::` suffix, so a formatted log line would read `INFO [trailing::] message`. Real `log::Record::target()` values never end in `::`, so this is cosmetic only, but if you want the fallback to look clean you could strip the separator:

```rust
pub(crate) fn target_label(target: &str) -> &str {
    match target.rsplit_once("::") {
        Some((head, "")) => head,
        Some((_, last)) => last,
        None => target,
    }
}
```

Safe to leave as-is.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+/// of the record's target module path.
+///
+/// # Panics
+///
+/// Panics if the logger cannot be built or if a global logger has already been set.
+pub(crate) fn init_logger() {
+    let logger = Logger::builder()
+        .default_endpoint("tslog")
+        .echo_stdout(true)
+        .max_level(log::LevelFilter::Info)
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:33]: 🌱 **seedling** — Hardcoded `LevelFilter::Info` is preserved from the previous inline version, so not a regression. Now that logging lives in its own module, wiring the level to `Settings` (or a `TS_LOG_LEVEL` env var / dictionary entry) would be a natural follow-up so log verbosity can be changed without a redeploy. Not for this PR.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        .build()
+        .expect("should build Logger");
+
+    fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!(
+                "{} {} [{}] {}",
+                Local::now().to_rfc3339_opts(SecondsFormat::Millis, true),
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: 🌱 **seedling** — Pre-existing behavior, moved verbatim — not a blocker. Under `wasm32-wasip1` Fastly Compute has no local timezone database, so `chrono::Local::now()` effectively resolves to UTC via chrono's WASI fallback. Consider switching to `chrono::Utc::now()` explicitly in a follow-up so the code is truthful about what it emits and doesn't rely on an implicit fallback.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                record.level(),
+                target_label(record.target()),
+                message
+            ));
+        })
+        .chain(Box::new(logger) as Box<dyn log::Log>)
+        .apply()
+        .expect("should initialize logger");
+}
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:50]: 👍 **praise** — Clean boundary move. `pub(crate)` on `init_logger`, idiomatic fern callback restored, `rsplit_once` with an explicit empty-segment fallback, and tests that pin all four edge cases. Exactly what the prior review asked for, done with minimal surface area. Nice execution on the second pass.

File: docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md [hunk 8]
```diff
@@ -0,0 +1,287 @@
+# PR 10 Logging Initialization Implementation Plan
```
Comment (aram356) [inline docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md:1]: ⛏ **nitpick** — Prior review flagged that this plan still showed `split("::").last().unwrap_or(target)` while the implementation switched to `rsplit_once`. Worth grepping the plan for that snippet and updating it so the plan doesn't drift from the shipped code — otherwise the doc becomes a stale artifact once this merges.

**Review by aram356 — APPROVED** [review]
## Summary

Second-pass review. Prior CHANGES_REQUESTED findings from 2026-04-10 are all resolved in commit dd6929c3: `#[allow(dead_code)]` removed, idiomatic fern `message` parameter restored, `rsplit_once` with empty-segment fallback in place, and `target_label` tests cover the no-separator, empty-string, and trailing-`::` cases. The core ↔ adapter boundary move is clean — `log-fastly` is gone from `trusted-server-core` and lives only in `trusted-server-adapter-fastly`. CI is green. Approving with a few non-blocking suggestions inline.

## Non-blocking

### ⛏ nitpick

- **`target_label` visibility**: could be private instead of `pub(crate)` (logging.rs:9)
- **Trailing-`::` fallback still contains the separator**: cosmetic only; real log targets never hit this (logging.rs:13)
- **Plan doc drift**: plan file still references the pre-`rsplit_once` snippet per the prior review

### 🌱 seedling

- **Hardcoded `LevelFilter::Info`**: natural follow-up to wire through `Settings` now that logging owns its own module (logging.rs:29)
- **`chrono::Local::now()` under WASI**: effectively resolves to UTC via fallback; consider switching to `chrono::Utc::now()` explicitly (logging.rs:37)

### 👍 praise

- **Clean boundary move**: minimal surface, focused tests, prior feedback fully addressed (logging.rs)

## CI Status

- browser integration tests: PASS
- integration tests: PASS
- prepare integration artifacts: PASS

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 9]
```diff
@@ -0,0 +1,77 @@
+use chrono::{SecondsFormat, Utc};
+use log_fastly::Logger;
+
+/// Extracts the final `::` segment from a Rust module path for use as a log label.
+///
+/// Falls back to the full target string when the input contains no separator or
+/// when the separator appears at the trailing position (e.g. `"foo::"`), which
+/// would otherwise produce an empty label in log output.
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: 🔧 **wrench** — Doc comment no longer matches behavior.

After commit `1acbfa7b`, the trailing-`::` case returns the *head* segment (stripping the separator), not the full target. The current doc still claims it "falls back to the full target string ... when the separator appears at the trailing position," but the test at line 72-75 asserts `"trailing::"` → `"trailing"`, which is a strip, not a fallback.

**Fix**:

```rust
/// Extracts the final `::` segment from a Rust module path for use as a log label.
///
/// When the input has no `::` separator, returns the full target. When the
/// separator is at the trailing position (e.g. `"foo::"`), returns the head
/// segment (`"foo"`) to avoid emitting an empty label.
```

File: docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+pub(crate) fn init_logger() {
+    let logger = Logger::builder()
+        .default_endpoint("tslog")
+        .echo_stdout(true)
+        .max_level(log::LevelFilter::Info)
+        .build()
+        .expect("should build Logger");
+
+    fern::Dispatch::new()
+        .format(|out, _message, record| {
```
Comment (aram356) [inline docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md]: 🔧 **wrench** — Plan drifts from the shipped implementation.

This block shows `|out, _message, record|` + `record.args()`, but `logging.rs:35` uses the idiomatic `|out, message, record|` + `message` — the form that commit `49e3f1d0` intentionally restored after prior review feedback. This drift was flagged in the 2026-04-10 and 2026-04-14 reviews and still hasn't been resolved.

**Fix**: update the block to mirror the shipped implementation, or drop the code snippet entirely. Plan docs that misdescribe shipped code are worse than no snippet.

```rust
.format(|out, message, record| {
    out.finish(format_args!(
        "{} {} [{}] {}",
        Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true),
        record.level(),
        target_label(record.target()),
        message
    ));
})
```

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+fn target_label(target: &str) -> &str {
+    match target.rsplit_once("::") {
+        Some((head, "")) => head,
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:13]: 🤔 **thinking** — `rsplit_once` only splits on the *last* `::`. For a hypothetical `"a::b::"`, the match yields `Some(("a::b", ""))` and returns `"a::b"` with the intermediate separator intact. Unlikely in practice (Rust module paths don't normally end with `::`), but the "strip separator" phrasing in the test assertion implies a cleaner rule than the code actually implements. Either tighten the doc to name the narrow guarantee, or iterate until no trailing `::` remains.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+/// of the record's target module path.
+///
+/// # Panics
+///
+/// Panics if the logger cannot be built or if a global logger has already been set.
+pub(crate) fn init_logger() {
+    let logger = Logger::builder()
+        .default_endpoint("tslog")
+        .echo_stdout(true)
+        .max_level(log::LevelFilter::Info)
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:33]: 🌱 **seedling** — `max_level(LevelFilter::Info)` is hardcoded. Now that logging is adapter-owned, `main.rs:54`'s `log::debug!("Settings {settings:?}")` is effectively dead (filtered out). Natural follow-up: wire through `Settings` so operators can opt into debug-level locally without a recompile.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 13]
```diff
@@ -0,0 +1,77 @@
+use chrono::{SecondsFormat, Utc};
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:3]: ⛏ **nitpick** — File has no module-level `//!` doc. A one-liner ("Fastly-specific logger wiring for the trusted-server adapter") would help navigation since there are no public symbols to document the file's purpose via rustdoc.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+/// Initialises the Fastly-backed `fern` logger and installs it as the global logger.
+///
+/// Log records are forwarded to the `tslog` Fastly endpoint and echoed to stdout.
+/// Each line is prefixed with an RFC 3339 timestamp, level, and the final segment
+/// of the record's target module path.
+///
+/// # Panics
+///
+/// Panics if the logger cannot be built or if a global logger has already been set.
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: ⛏ **nitpick** — Panics phrasing is slightly off. `Logger::builder().build()` returning `Err` is practically unreachable at startup; the real panic source is `fern::Dispatch::apply()` double-install. Could be tightened to name the real failure mode, but fine as-is.

File: crates/trusted-server-adapter-fastly/src/logging.rs [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+                Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true),
+                record.level(),
+                target_label(record.target()),
+                message
+            ));
+        })
+        .chain(Box::new(logger) as Box<dyn log::Log>)
+        .apply()
+        .expect("should initialize logger");
+}
```
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:50]: 👍 **praise** — Three rounds of review feedback cleanly addressed in the final commit: visibility tightened (`pub(crate)` → private), `Utc::now()` replaces `Local::now()` (correct under WASI), and trailing-`::` now strips the separator with matching test coverage. No scope creep. Nice work.

**Review by aram356 — DISMISSED** [review]
## Summary

Deep re-review of the PR10 logging-boundary extraction. The core change — moving Fastly logger setup into an adapter-local module and dropping `log-fastly` from `trusted-server-core` — is well-scoped, CI is green, and three prior rounds of review feedback have been cleanly resolved in `1acbfa7b` (visibility tightened, `Utc::now()` under WASI, trailing-`::` strip with matching test coverage).

Requesting changes on two items that are small but should land before merge:

1. The `target_label` doc comment now describes the pre-`1acbfa7b` behavior and contradicts the test assertion.
2. The plan doc's code snippet still drifts from the shipped implementation — flagged in the 2026-04-10 and 2026-04-14 reviews, still unresolved.

Everything else is seedling / nitpick / praise (inline).

**Scope note**: This PR stacks on `feature/edgezero-pr9-wire-signing-to-store-primitives`, not `main`. Review covers the 6-file / 437-line delta against the stated base branch. A full-stack audit vs `main` would span PR 2-9 and is not in this review's scope.

## Blocking

### 🔧 wrench

- **Doc/behavior mismatch on `target_label`**: doc says trailing-`::` falls back to the full target, but implementation (and test) strips the separator (`logging.rs:6-8`)
- **Plan doc still drifts**: snippet uses `|out, _message, record|` + `record.args()` but shipped code uses `message`; noted twice in prior reviews (`docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md:97`)

## Non-blocking

### 🤔 thinking

- **`target_label(\"a::b::\")` behavior**: `rsplit_once` keeps intermediate `::` in the head; edge case unlikely in real module paths but worth documenting or iterating (`logging.rs:11`)

### 🌱 seedling / 📌 out of scope

- **`max_level` hardcoded**: wire through `Settings` as a follow-up so `log::debug!` in `main.rs:54` isn't silently filtered (`logging.rs:30`)
- **Core still depends on `fastly`**: `log-fastly` removal is a step forward but doesn't finish the platform-agnostic core goal. `trusted-server-core/Cargo.toml` still has `fastly = { workspace = true }`. Worth tracking as a follow-up issue after the PR 2-9 stack lands.

### ⛏ nitpick

- **Missing module-level `//!` doc** on `logging.rs` (`logging.rs:1`)
- **`# Panics` phrasing** names logger build failure, but real panic source is double-`apply()` (`logging.rs:25`)

### 👍 praise

- **Clean resolution of three review rounds** — no scope creep, matching test coverage for each behavior change (`logging.rs:47`)

## CI Status

- integration tests: PASS
- browser integration tests: PASS
- prepare integration artifacts: PASS
- local `cargo test -p trusted-server-adapter-fastly`: 12/12 PASS on `wasm32-wasip1`

File: crates/trusted-server-adapter-fastly/src/logging.rs (hunk 9 above)
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: 🔧 **wrench** — Doc comment no longer matches behavior.

After commit `1acbfa7b`, the trailing-`::` case returns the *head* segment (stripping the separator), not the full target. The current doc still claims it "falls back to the full target string ... when the separator appears at the trailing position," but the test at line 72-75 asserts `"trailing::"` → `"trailing"`, which is a strip, not a fallback.

**Fix**:

```rust
/// Extracts the final `::` segment from a Rust module path for use as a log label.
///
/// When the input has no `::` separator, returns the full target. When the
/// separator is at the trailing position (e.g. `"foo::"`), returns the head
/// segment (`"foo"`) to avoid emitting an empty label.
```

File: docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md (hunk 10 above)
Comment (aram356) [inline docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md]: 🔧 **wrench** — Plan drifts from the shipped implementation.

This block shows `|out, _message, record|` + `record.args()`, but `logging.rs:35` uses the idiomatic `|out, message, record|` + `message` — the form that commit `49e3f1d0` intentionally restored after prior review feedback. This drift was flagged in the 2026-04-10 and 2026-04-14 reviews and still hasn't been resolved.

**Fix**: update the block to mirror the shipped implementation, or drop the code snippet entirely. Plan docs that misdescribe shipped code are worse than no snippet.

```rust
.format(|out, message, record| {
    out.finish(format_args!(
        "{} {} [{}] {}",
        Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true),
        record.level(),
        target_label(record.target()),
        message
    ));
})
```

File: crates/trusted-server-adapter-fastly/src/logging.rs (hunk 11 above)
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:13]: 🤔 **thinking** — `rsplit_once` only splits on the *last* `::`. For a hypothetical `"a::b::"`, the match yields `Some(("a::b", ""))` and returns `"a::b"` with the intermediate separator intact. Unlikely in practice (Rust module paths don't normally end with `::`), but the "strip separator" phrasing in the test assertion implies a cleaner rule than the code actually implements. Either tighten the doc to name the narrow guarantee, or iterate until no trailing `::` remains.

File: crates/trusted-server-adapter-fastly/src/logging.rs (hunk 12 above)
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:33]: 🌱 **seedling** — `max_level(LevelFilter::Info)` is hardcoded. Now that logging is adapter-owned, `main.rs:54`'s `log::debug!("Settings {settings:?}")` is effectively dead (filtered out). Natural follow-up: wire through `Settings` so operators can opt into debug-level locally without a recompile.

File: crates/trusted-server-adapter-fastly/src/logging.rs (hunk 13 above)
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs:3]: ⛏ **nitpick** — File has no module-level `//!` doc. A one-liner ("Fastly-specific logger wiring for the trusted-server adapter") would help navigation since there are no public symbols to document the file's purpose via rustdoc.

File: crates/trusted-server-adapter-fastly/src/logging.rs (hunk 14 above)
Comment (aram356) [inline crates/trusted-server-adapter-fastly/src/logging.rs]: ⛏ **nitpick** — Panics phrasing is slightly off. `Logger::builder().build()` returning `Err` is practically unreachable at startup; the real panic source is `fern::Dispatch::apply()` double-install. Could be tightened to name the real failure mode, but fine as-is.

**Review by aram356 — CHANGES_REQUESTED** [review]
## Summary

Deep re-review of the PR10 logging-boundary extraction. The core change — moving Fastly logger setup into an adapter-local module and dropping `log-fastly` from `trusted-server-core` — is well-scoped and CI is green.

Requesting changes on two items that are small but should land before merge:

1. The `target_label` doc comment now describes the pre-`1acbfa7b` behavior and contradicts the test assertion.
2. The plan doc's code snippet still drifts from the shipped implementation — flagged in the 2026-04-10 and 2026-04-14 reviews, still unresolved.

Everything else is seedling / nitpick / praise (inline).

**Scope note**: This PR stacks on `feature/edgezero-pr9-wire-signing-to-store-primitives`, not `main`. Review covers the 6-file / 437-line delta against the stated base branch. A full-stack audit vs `main` would span PR 2-9 and is not in this review's scope.

## Blocking

### 🔧 wrench

- **Doc/behavior mismatch on `target_label`**: doc says trailing-`::` falls back to the full target, but implementation (and test) strips the separator (`logging.rs:6-8`)
- **Plan doc still drifts**: snippet uses `|out, _message, record|` + `record.args()` but shipped code uses `message`; noted twice in prior reviews (`docs/superpowers/plans/2026-04-02-pr10-logging-initialization.md:97`)

## Non-blocking

### 🤔 thinking

- **`target_label(\"a::b::\")` behavior**: `rsplit_once` keeps intermediate `::` in the head; edge case unlikely in real module paths but worth documenting or iterating (`logging.rs:11`)

### 🌱 seedling / 📌 out of scope

- **`max_level` hardcoded**: wire through `Settings` as a follow-up so `log::debug!` in `main.rs:54` isn't silently filtered (`logging.rs:30`)
- **Core still depends on `fastly`**: `log-fastly` removal is a step forward but doesn't finish the platform-agnostic core goal. `trusted-server-core/Cargo.toml` still has `fastly = { workspace = true }`. Worth tracking as a follow-up issue after the PR 2-9 stack lands.

### ⛏ nitpick

- **Missing module-level `//!` doc** on `logging.rs` (`logging.rs:1`)
- **`# Panics` phrasing** names logger build failure, but real panic source is double-`apply()` (`logging.rs:25`)

## CI Status

- integration tests: PASS
- browser integration tests: PASS
- prepare integration artifacts: PASS
- local `cargo test -p trusted-server-adapter-fastly`: 12/12 PASS on `wasm32-wasip1`

**Review by aram356 — APPROVED** [review]
👍 Looks good
